socialstream/chat/chat.py

```diff
@@ -159,242 +225,334 @@ def get_env_agents(env_agent_combo) -> None:
+            )
 
-    initialize_session_state()
+        agent_col1, agent_col2 = st.columns(2)
+        with agent_col1:
+            agent_choice_1 = st.selectbox(
+                "Choose the first agent:",
+                agent_list_1.keys(),
+                disabled=st.session_state.active,
+                index=0,
+            )
+        with agent_col2:
+            agent_choice_2 = st.selectbox(
+                "Choose the second agent:",
+                agent_list_2.keys(),
+                disabled=st.session_state.active,
+                index=1,
+            )
+        print(agent_choice_1, agent_choice_2)
+        user_position = st.selectbox(
+            "Which agent do you want to be?",
+            [agent_choice_1, agent_choice_2],
+            disabled=st.session_state.active,
+        )
+        # user_position = st.selectbox(
+        #     "Do you want to be the first or second agent?",
+        #     POSITION_CHOICES,
+        #     disabled=st.session_state.active,
+        # )
+        if agent_choice_1 == agent_choice_2:
+            st.warning(
+                "The two agents cannot be the same. Please select different agents."
+            )
+            st.stop()
 
-    scenarios = st.session_state.env_mapping
-    agent_list_1, agent_list_2 = st.session_state.agent_mapping
+        if (
+            agent_choice_1
+            or agent_choice_2
+            or user_position
+            or scenario_choice
+            or model_choice
+        ) and not st.session_state.active:
+            global MODEL
+            MODEL = model_choice
+            print(f"Setting settings with model {MODEL}...")
+            set_settings(
+                agent_choice_1,
+                agent_choice_2,
+                scenario_choice,
+                user_position,
+                [agent_choice_1, agent_choice_2],
+            )
 
-    scenario_choice = st.selectbox(
-        "Choose a scenario:", scenarios.keys(), disabled=st.session_state.active
-    )
+    with st.expander("Check your social task!", expanded=True):
+        agent_infos = compose_agent_messages()
+        env_info, goals_info = compose_env_messages()
 
-    agent_col1, agent_col2 = st.columns(2)
-    with agent_col1:
-        agent_choice_1 = st.selectbox(
-            "Choose the first agent:",
-            agent_list_1.keys(),
-            disabled=st.session_state.active,
+        st.markdown(
+            f"""**Scenario:** {env_info}""",
+            unsafe_allow_html=True,
         )
-    with agent_col2:
-        agent_choice_2 = st.selectbox(
-            "Choose the second agent:",
-            agent_list_2.keys(),
-            disabled=st.session_state.active,
+
+        agent1_col, agent2_col = st.columns(2)
+        agent_cols = [agent1_col, agent2_col]
+        for agent_idx, agent_info in enumerate(agent_infos):
+            agent_col = agent_cols[agent_idx]
+            with agent_col:
+                st.markdown(
```

Comment:
st.markdown (pure text but with format) or st.text_area (with border, but no md format)?

Reply:
？are we using chat_new or chat?

Reply:
I remove chat.py and rename chat_new.py